samples/file_sdk_sample/main.cpp

```diff
@@ -0,0 +1,178 @@
+#include "main.h"
+#include <iostream>
+#include <memory>
+#include <fstream>
+
+#include <cxxopts/cxxopts.hpp>
+#include <Common/file_format_factory.h>
+#include <Common/std_stream_adapter.h>
+#include <api/istream_handler.h>
+#include <Common/tag.h>
+#include <CryptoAPI.h>
+#include "utils.h"
+
+using std::cout;
+using mip::file::StdStreamAdapter;
+using mip::Tag;
+
+namespace {
+Tag kGeneral("{434fea85-168e-4781-8780-6f89f3887290}", "General", "", "Assaf", true, "now", mip::Method::NONE);
+Tag kPersonal("{23a793b5-66c0-4dec-adb6-7928bb6df345}", "Personal", "", "Assaf", true, "now", mip::Method::MANUAL);
+Tag kSecret("{ccf69ce5-ac74-43e9-ba46-eb73ed3cac1b}", "Secret", "", "Assaf", true, "now", mip::Method::MANUAL);
+std::vector<Tag> kLabels { kGeneral, kPersonal, kSecret};
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    cxxopts::Options options(argv[0], "Sample app for Microsoft Information Protection SDK");
+    options.positional_help("<Extra args>");
+
+    options.add_options()
+        ("f,file", "Full path to the file to work on.", cxxopts::value<std::string>(), "File path")
+        ("g,get-label", "Show the labels that applies on a file.")
+        ("s,set-label", "Set a label with <labelId> and with owner <owner>. "
+                        "If downgrading label - will apply <Justification message> as needed.", cxxopts::value<std::vector<std::string>>())
+        ("r,remove-label", "Remove the current label from the file with <Justification message>.", cxxopts::value<std::string>())
+        ("l,list-labels", "Show all available labels with there Id's.")
+        ("h,help", "Print help and exit.")
+        ;
```

Comment:
show this to Eyal

Reply:
why? this is not a product thing. it is not the design of the API. it's just a sample


---
In reply to: [139348553](https://github.com/AzureAD/rms-sdk-for-cpp/pull/153#discussion_r139348553) [](ancestors = 139348553)